Offline mode (closes #1181)

george-gca opened the pull request:
Created a plugin to tackle #1181. Currently have an issue with tikzjax since it imports some wasm file from its javascript. The rest should work as expected.

alshedivat commented:
@george-gca, looks like this PR adds multiple plugins / modifies a few. could you elaborate on all added modifications?

george-gca commented:
It actually only adds one plug-in. The changes made to the others was just about indentation, to keep a standard. Some of them used 4 spaces, others 2.

george-gca commented:
I also renamed one plugin file to keep it consistent with the others. And moved almost all urls from libraries to `_config.yml` to allow downloading them from the new plugin in `_plugins/download-3rd-party.rb`
